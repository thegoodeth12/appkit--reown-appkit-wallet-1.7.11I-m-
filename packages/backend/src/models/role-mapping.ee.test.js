import { describe, it, expect } from 'vitest';
import RoleMapping from '@/models/role-mapping.ee.js';
import SamlAuthProvider from '@/models/saml-auth-provider.ee.js';
import Base from '@/models/base.js';

describe('RoleMapping model', () => {
  it('tableName should return correct name', () => {
    expect(RoleMapping.tableName).toBe('role_mappings');
  });

  it('jsonSchema should have the correct schema', () => {
    expect(RoleMapping.jsonSchema).toMatchSnapshot();
  });

  it('relationMappings should return correct associations', () => {
    const relationMappings = RoleMapping.relationMappings();

    const expectedRelations = {
      samlAuthProvider: {
        relation: Base.BelongsToOneRelation,
        modelClass: SamlAuthProvider,
        join: {
          from: 'role_mappings.saml_auth_provider_id',
          to: 'saml_auth_providers.id',
        },
      },
    };

    expect(relationMappings).toStrictEqual(expectedRelations);
  });
});
