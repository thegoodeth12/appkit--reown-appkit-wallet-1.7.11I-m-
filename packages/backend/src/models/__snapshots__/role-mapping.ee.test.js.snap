// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RoleMapping model > jsonSchema should have the correct schema 1`] = `
{
  "properties": {
    "id": {
      "format": "uuid",
      "type": "string",
    },
    "remoteRoleName": {
      "minLength": 1,
      "type": "string",
    },
    "roleId": {
      "format": "uuid",
      "type": "string",
    },
    "samlAuthProviderId": {
      "format": "uuid",
      "type": "string",
    },
  },
  "required": [
    "samlAuthProviderId",
    "roleId",
    "remoteRoleName",
  ],
  "type": "object",
}
`;
